tests/test_inputmedia.py

```diff
@@ -443,6 +443,64 @@ async def test_send_media_group_photo(self, bot, chat_id, media_group):
             mes.caption_entities == [MessageEntity(MessageEntity.BOLD, 0, 5)] for mes in messages
         )
 
+    async def test_send_media_group_throws_error_with_group_caption_and_individual_captions(
+        self, bot, chat_id, media_group
+    ):
+        with pytest.raises(
+            ValueError, match="You can only supply either group caption or media with captions."
+        ):
+            await bot.send_media_group(chat_id, media_group, caption="foo")
+
+    @pytest.mark.parametrize(
+        "caption, parse_mode, caption_entities",
+        [
+            # same combinations of caption options as in media_group fixture
+            ("*photo* 1", "Markdown", None),
+            ("<b>photo</b> 1", "HTML", None),
+            ("photo 1", None, [MessageEntity(MessageEntity.BOLD, 0, 5)]),
+        ],
+    )
+    @flaky(3, 1)
+    async def test_send_media_group_with_group_caption(
+        self,
+        bot,
+        chat_id,
+        media_group,
+        caption,
+        parse_mode,
+        caption_entities,
+    ):
+        # clear media_group of all_captions
+        for item in media_group:
+            item.caption = None
+            item.parse_mode = None
+            item.caption_entities = None
+
+        messages = await bot.send_media_group(
+            chat_id,
+            media_group,
+            caption=caption,
+            parse_mode=parse_mode,
+            caption_entities=caption_entities,
+        )
+        assert isinstance(messages, list)
+        assert len(messages) == 3
+        assert all(isinstance(mes, Message) for mes in messages)
+
+        first_message, other_messages = messages[0], messages[1:]
+        assert all(mes.media_group_id == first_message.media_group_id for mes in messages)
+
+        # Make sure first message got the caption, which will lead
+        # to Telegram displaying its caption as group caption
+        assert first_message.caption
+        assert first_message.caption_entities == [MessageEntity(MessageEntity.BOLD, 0, 5)]
+
+        # Check that other messages have no captions
+        assert all(mes.caption is None for mes in other_messages)
+        assert not any(mes.caption_entities for mes in other_messages)
+        # .parse_mode must be completely absent
+        assert not any(getattr(mes, "parse_mode", None) for mes in other_messages)
```

Comment:
the `Message` class doesn't have a `parse_mode` attribute, so this assertion doesn't really help. You already check that the `caption_entities` are empty - that suffices

Reply:
got it